### PR TITLE
Key Vault keys API: Add OKP key type and EdDSA digital signature

### PR DESCRIPTION
**Pull request by docschmidt**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
 
  The feature is already deployed to public regions.

  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.

Expectation is to release with the next preview version of the Key Vault data plane APIs.

  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json [hunk 1]
```diff
@@ -1352,6 +1354,14 @@
...
                 "value": "oct-HSM",
                 "description": "Octet sequence (used to represent symmetric keys) which is stored the HSM."
+              },
+              {
+                "value": "OKP",
+                "description": "Octet key pair (https://tools.ietf.org/html/rfc8037)"
+              },
+              {
+                "value": "OKP-HSM",
+                "description": "Octet key pair (https://tools.ietf.org/html/rfc8037) with a private key which is stored in the HSM."
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json:1364]: While the descriptions seem fine, I'm wondering if this support extends to Key Vault or only Managed HSM? In the latter case, isn't this description always true then?

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json [hunk 2]
```diff
@@ -1445,14 +1456,18 @@
               {
                 "value": "P-256K",
                 "description": "The SECG SECP256K1 elliptic curve."
+              },
+              {
+                "value": "Ed25519",
+                "description": "The Ed25519 Edwards-curve."
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json:1462]: Nit: when not used as a compound adjective, it's just "Edwards curve". RFCs most often use it as two distinct words as well.

Comment (heaths) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json:1462]: If you change it, there's a couple instances throughout. Please be sure to update them all.

**Review by heaths — APPROVED** [review]

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json (hunk 1 above)
Comment (docschmidt) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json:1364]: This key type is only supported for Managed HSM. So, OKP and OKP-HSM are synonymous and both use the HSM for storing the keys. I have no insights into whether Key Vault will support this key type eventually. 

I followed the example of the `oct` key type which AFAIK is also only supported by Managed HSM.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json (hunk 2 above)
Comment (docschmidt) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/keys.json:1462]: Will update in next iteration.

**Review by heaths — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@docschmidt can you update your branch with the latest content from `main`?  I believe that will fix the hanging CI problem.

**Comment by docschmidt** [comment]
> @docschmidt can you update your branch with the latest content from `main`? I believe that will fix the hanging CI problem.

Will do.

**Comment by docschmidt** [comment]
> @docschmidt can you update your branch with the latest content from `main`? I believe that will fix the hanging CI problem.

@jhendrixMSFT, looks like merging `main` brought in other issues. Are you aware of a more specific update I should merge? Otherwise does it make sense to merge?

**Comment by heaths** [comment]
It's going to have to merge with `main` eventually so it's best to go through the issues and fix what you can. The breaking change policy can be ignored, and will have to. But things like PrettierCheck are typically pretty straight-forward to resolve.

**Comment by docschmidt** [comment]
@heaths and @jhendrixMSFT,

I reviewed the failed checks and none of them are under `specification/keyvault`. Is the expectation still to try to fix those? Wouldn't this trigger reviews from those teams?

**Review by heaths — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by heaths** [comment]
When you do, please rebase - not merge - main:

```bash
git fetch upstream main
git rebase upstream/main
```

**Comment by docschmidt** [comment]
As I understand it, the problem might be that the target branch (Azure:dev-keyvault-Microsoft.KeyVault-7.4-preview) is too far behind. If that is the case, then rebasing my feature branch will bring in again all these changes from main that may or may not work.

Wouldn't it make sense to rebase `Azure:dev-keyvault-Microsoft.KeyVault-7.4-preview` first?
________________________________

**Comment by heaths** [comment]
Actually, I missed this was targeting a release branch. Don't rebase until we're done and ready to merge, or anyone else with outstanding work will be adversely affected.

Looking at the checks, this pipeline did pass but didn't report back status: https://github.com/Azure/azure-rest-api-specs/pull/15989/checks?check_run_id=4002543967

Given that, @lmazuel can you help us merge this?

**Comment by docschmidt** [comment]
@lmazuel, gentle ping on @heaths' ask. 

Could you please help merge the change into the feature branch?

**Pull request merged**